src/main/java/com/app/onlinebookstore/specifications/BookSpecifications.java

```diff
@@ -0,0 +1,34 @@
+package com.app.onlinebookstore.specifications;
+
+import com.app.onlinebookstore.dto.BookSearchParametersDto;
+import com.app.onlinebookstore.model.Book;
+import jakarta.persistence.criteria.Predicate;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.data.jpa.domain.Specification;
+
+public class BookSpecifications {
+    public static Specification<Book> searchBooks(BookSearchParametersDto searchParameters) {
+        return (root, query, cb) -> {
```

Comment:
I think it's a more understandable name of argument.
```suggestion
        return (root, query, criteriaBuilder) -> {
```